ts/editor/ClozeButton.svelte

```diff
@@ -5,20 +5,29 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 <script lang="typescript">
     import IconButton from "../components/IconButton.svelte";
     import WithShortcut from "../components/WithShortcut.svelte";
-    import OnlyEditable from "./OnlyEditable.svelte";
 
     import * as tr from "../lib/ftl";
     import { withButton } from "../components/helpers";
     import { ellipseIcon } from "./icons";
-    import { forEditorField } from ".";
-    import { wrapCurrent } from "./wrap";
+    import { get } from "svelte/store";
+    import {
+        getContext,
+        noteEditorKey,
+        activeInputKey,
+        focusInEditableKey,
+    } from "./context";
+
+    const noteEditor = getContext(noteEditorKey);
+    const activeInput = getContext(activeInputKey);
 
     const clozePattern = /\{\{c(\d+)::/gu;
     function getCurrentHighestCloze(increment: boolean): number {
         let highest = 0;
 
-        forEditorField([], (field) => {
-            const fieldHTML = field.editingArea.fieldHTML;
+        for (const field of noteEditor.fields) {
+            const content = field.editingArea?.content;
+            const fieldHTML = content ? get(content) : "";
```

Comment:
Exporting the actual value of the store, we would have to export a function, like so:

```
Object.assign(api, {
    $content: () => $content,
})
```

Which would turn `get(component.subcomponent.value).subvalue` into `component.subcomponent.$value().subvalue`. I wasn't sure whether this is better, as it's somewhat obfuscated. It also increases the API surface by a bit.

Maybe we could write an extended version of the svelte built-in `writable`, and create a `gettable`, or I guess a `readGettable` and a `writeGettable`, which expose a `.get()` method, rather than having to use (and import) a function.

Actually another issue intertwined here is that of `.editingArea`. I exposed it as an optional property, but it would probably _more technically correctly_ but also maybe better to put expose it as a promise, because at the time `EditorField` sets the context, it is not available once it mounts. Especially if you access the component across other components, this might be an issue, as it's not always clear in which order components mount. Using a promise would turn the above code into:
```js
Promise.all(noteEditor.fields.map((field) => field.editingArea)).then(editingAreas => editingAreas.map(earea => get(earea.content)))
```
... Phew

Reply:
> Which would turn get(component.subcomponent.value).subvalue into component.subcomponent.$value().subvalue.

Ah, that seems worse.

> .. Phew

Indeed :-) If it's something that needs to be done regularly, maybe the top level editor could provide some convenience accessors so callers can do something like `await noteEditor.editingAreaForField(x)`? Not a recommendation, just brainstorming.

Reply:
I agree some convenience accessors might be nice. But I'd prefer to implement these while also exploring writing add-ons with it myself. Otherwise it's hard to make judgements on what _might_ be useful.

Reply:
Yep, totally agree